uefi/src/proto/device_path/mod.rs

```diff
@@ -266,6 +266,22 @@ impl PartialEq for DevicePathNode {
     }
 }
 
+impl<'a> TryFrom<&[u8]> for &'a DevicePathNode {
+    type Error = ByteConversionError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if mem::size_of::<DevicePathHeader>() <= bytes.len() {
```

Comment:
I don't think this initial check is needed now, `<&DevicePathHeader>::try_from` already checks it.